olp-cpp-sdk-core/include/olp/core/thread/ThreadPoolTaskScheduler.h

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (C) 2019 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+#pragma once
+
+#include <thread>
+#include <vector>
+
+#include <olp/core/thread/SyncQueue.h>
+#include <olp/core/thread/TaskScheduler.h>
+
+namespace olp {
+namespace thread {
+
+class CORE_API ThreadPoolTaskScheduler final : public TaskScheduler {
```

Comment:
What about copy/move operations? We should rather support them or disable them in TaskScheduler class.

Reply:
As the TaskScheduler is an "abstract" interface there is not much need for deleting, I would leave to the implementation to decide if he supports copy & move operations.
For the ThreadPoolTaskScheduler it is implicitly deleted, both copy & move operations, as I have a user-declared destructor and SyncQueue is not copiable. But to be cleared i will add them.